src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/AdminController.kt

```diff
@@ -0,0 +1,76 @@
+package uk.gov.justice.digital.hmpps.personrecord.api.controller
+
+import io.swagger.v3.oas.annotations.Hidden
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.personrecord.api.model.admin.AdminReclusterRecord
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonEntity
+import uk.gov.justice.digital.hmpps.personrecord.jpa.repository.PersonRepository
+import uk.gov.justice.digital.hmpps.personrecord.model.types.SourceSystemType
+import uk.gov.justice.digital.hmpps.personrecord.service.cprdomainevents.events.telemetry.RecordClusterTelemetry
+import uk.gov.justice.digital.hmpps.personrecord.service.message.recluster.ReclusterService
+import uk.gov.justice.digital.hmpps.personrecord.service.search.PersonMatchService
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType
+
+@RestController
+@RequestMapping("/admin")
+class AdminController(
+  private val reclusterService: ReclusterService,
+  private val personRepository: PersonRepository,
+  private val personMatchService: PersonMatchService,
+  private val publisher: ApplicationEventPublisher,
+) {
+
+  @Hidden
+  @PostMapping("/recluster")
+  suspend fun postRecluster(
+    @RequestBody adminReclusterRecords: List<AdminReclusterRecord>,
+  ) {
+    CoroutineScope(Dispatchers.Default).launch {
+      log.info("$RECLUSTER_PROCESS_PREFIX Triggered. Number of records: ${adminReclusterRecords.size}.")
+      upsertRecords(adminReclusterRecords)
+      log.info("$RECLUSTER_PROCESS_PREFIX Records Upsert Complete.")
+      triggerRecluster(adminReclusterRecords)
+      log.info("$RECLUSTER_PROCESS_PREFIX Complete.")
+    }
+  }
+
+  private fun upsertRecords(adminReclusterRecords: List<AdminReclusterRecord>) {
+    adminReclusterRecords.forEach {
+      searchForPersonByIdentifier(it)?.let { person ->
+        personMatchService.saveToPersonMatch(person)
+      }
+    }
+  }
+
+  private fun triggerRecluster(adminReclusterRecords: List<AdminReclusterRecord>) {
+    adminReclusterRecords.forEach {
+      searchForPersonByIdentifier(it)?.let { person ->
+        person.personKey?.let { cluster ->
+          publisher.publishEvent(RecordClusterTelemetry(TelemetryEventType.CPR_ADMIN_RECLUSTER_TRIGGERED, cluster))
+          reclusterService.recluster(cluster, person)
```

Comment:
what if the cluster is NEEDS_ATTENTION here according to processing with the incorrect model? Should we do the self-heal check?